lib/iris/tests/unit/fileformats/netcdf/test_Saver__ugrid.py

```diff
@@ -932,44 +932,52 @@ def _namestext(names):
         ]
         return f'({" ".join(name_texts)})'
 
-    def test_mesh_names__varname(self):
+    def test_mesh_names(self):
         # Check the selection of mesh-variables names.
         # N.B. this is basically centralised in Saver._get_mesh_variable_name,
         # but we test in an implementation-neutral way (as it's fairly easy).
         mesh_names_tests = [
+            # no names : based on dimensionality
+            (
+                (None, None, None),
+                (None, None, "Mesh_2d"),
+            ),
             # var_name only
             (
                 (None, None, "meshvar_x"),
                 (None, None, "meshvar_x"),
             ),
-            # standard_name only
+            # standard_name only : does not apply to Mesh
             (
                 ("air_temperature", None, None),
-                ("air_temperature", None, "air_temperature"),
+                ("air_temperature", None, "Mesh_2d"),
             ),
             # long_name only
             (
                 (None, "my_long_name", None),
                 (None, "my_long_name", "my_long_name"),
             ),
+            # long_name that needs "fixing"
```

Comment:
Whoops.  I think we are now testing this twice.

Reply:
(removed now : https://github.com/SciTools/iris/pull/4318/commits/9ae1ea39e58dd7ed2eced2e9a58cef78822e9670)